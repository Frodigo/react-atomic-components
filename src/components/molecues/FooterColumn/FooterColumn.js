import React, { useState } from 'react';
import PropTypes from 'prop-types';

import PlusIcon from '../../../assets/icons/plus.svg';
import MinusIcon from '../../../assets/icons/minus.svg';
import { Link } from '../../atoms';

import classes from './FooterColumn.module.css';

const FooterColumn = (props) => {
    const { heading, subheadings, ...rest } = props;
    console.log(subheadings)
    const [isOpen, setIsOpen] = useState(false);

    const footerColumnClass = isOpen
        ? classes.footerColumnOpened
        : classes.footerColumn;
    const buttonIcon = isOpen ? <MinusIcon /> : <PlusIcon />;
    const linksContent = subheadings.map((item) => (

        <h6 key={item._uid} className={classes.subheading}>
            <Link className={classes.subheadingLink} href={item.url}>
                {item.label}
            </Link>
        </h6>
    ));

    return (
        <div className={footerColumnClass} {...rest}>
            <button
                type="button"
                onClick={() => setIsOpen(!isOpen)}
                className={classes.header}
            >
                <h5 className={classes.heading}>{heading}</h5>
                <span className={classes.expandButton}>{buttonIcon}</span>
            </button>
            <section className={isOpen ? classes.linksOpen : classes.links}>{linksContent}</section>
        </div>
    );
};

FooterColumn.propTypes = {
    heading: PropTypes.string.isRequired,
    rest: PropTypes.arrayOf(PropTypes.object),
    subheadings: PropTypes.array
};

export default FooterColumn;
