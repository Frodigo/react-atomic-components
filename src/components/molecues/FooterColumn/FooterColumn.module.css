.footerColumn {
    background-color: var(--color-alternate);
    border-bottom: solid 1px var(--color-secondary);
}

.footerColumnOpened {
    composes: footerColumn;
}

.footerColumnOpened .links {
    display: block;
}

.header {
    align-items: center;
    background: none;
    border: 0;
    cursor: pointer;
    display: flex;
    justify-content: space-between;
    min-height: 55px;
    padding: var(--spacing-xs);
    outline: none;
    width: 100%;
}

.heading {
    color: var(--color-white);
    margin: 0;
}


.links {
    display: none;
}

.linksOpen {
    display: block;
}

.expandButton {
    display: inline-block;
}

.subheading {
    border-top: 1px solid var(--color-gray-light);
}

.subheadingLink {
    color: var(--color-gray-light);
    display: block;
    padding: var(--spacing-xs);
    text-decoration: none;
}


@media (min-width: 1024px) {  
    .expandButton {
        display: none;
    }

    .header {
        cursor: text;
    }

    .footerColumn {
        flex: 1 1 20%;
        padding: var(--spacing-md);
    }

    .footerColumn .links {
        display: block;
    }

    .subheading {
        margin: 0;
    }

    .subheading:first-child {
        border-top: 0;
    }

    .subheadingLink {
        padding-bottom: var(--spacing-md);
    }
}

@media (min-width: 1300px) {
    .subheadingLink:hover,
    .subheadingLink:focus {
        text-decoration: underline;
    }
}