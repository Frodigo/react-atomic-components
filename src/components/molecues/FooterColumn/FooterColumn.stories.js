import React from 'react';

import { default as FooterColumnComponent } from './FooterColumn';

export default {
    title: 'Molecues/FooterColumn',
    component: FooterColumnComponent
};

const isOpen = false;
const Template = (args) => <FooterColumnComponent {...args} />;

export const Default = Template.bind({});
Default.args = {
    heading: 'Heading',
    subheadings: [
        {
            label: 'Subheading 1',
            url: '#'
        },
        {
            label: 'Subheading 2',
            url: '#'
        },
        {
            label: 'Subheading 3',
            url: '#'
        }
    ],
    isOpen,
    handleToggle: () => !isOpen
};
