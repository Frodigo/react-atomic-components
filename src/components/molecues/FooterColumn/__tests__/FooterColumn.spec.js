import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import FooterColumn from '../FooterColumn';

describe('FooterColumn should be rendered', () => {
    const columnMock = {
        heading: 'Heading',
        subheadings: [
            {
                label: 'Subheading 1',
                url: '#'
            },
            {
                label: 'Subheading 2',
                url: '#'
            },
            {
                label: 'Subheading 3',
                url: '#'
            }
        ]
    };

    test('FooterColumn renders correctly', () => {
        const { container } = render(
            <FooterColumn
                heading={columnMock.heading}
                subheadings={columnMock.subheadings}
            />
        );

        expect(container).toMatchSnapshot();
    });

    test('FooterColumn should be toggled', () => {
        const { getByText, queryByText } = render(
            <FooterColumn
                heading={columnMock.heading}
                subheadings={columnMock.subheadings}
            />
        );

        userEvent.click(screen.getByText('Heading'));
        getByText('Subheading 2')

        userEvent.click(screen.getByText('Heading'));
        expect(queryByText('Subheading 2')).toBeNull();
    });
});
