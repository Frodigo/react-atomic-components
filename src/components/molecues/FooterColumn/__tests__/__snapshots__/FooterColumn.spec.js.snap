// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterColumn should be rendered FooterColumn renders correctly 1`] = `
<div>
  <div
    class="footerColumn"
  >
    <button
      class="header"
      type="button"
    >
      <h5
        class="heading"
      >
        Heading
      </h5>
      <span
        class="expandButton"
      >
        <span>
          svg
        </span>
      </span>
    </button>
  </div>
</div>
`;
